### PR TITLE
fix(@angular-ru/ng-excel-builder): fixed automatic column width

### PR DESCRIPTION
**Pull request by StepaZol**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/ng-excel-builder/integration/tests/helpers/file-suites/test-4-options.xls [hunk 1]
```diff
@@ -1,131 +1,56 @@
...
-                    <Data ss:Type="String">ru</Data>
-                </Cell>
-                <Cell ss:StyleID="BodyCellStyle">
-                    <Data ss:Type="String">ru</Data>
-                </Cell>
-            </Row>
-        </Table>
-    </Worksheet>
-</Workbook>
+﻿
```
Comment (Schirbak) [inline packages/ng-excel-builder/integration/tests/helpers/file-suites/test-4-options.xls:1]: you can prettify test xmls for better review, test suites minify all files so they pass successfully

File: packages/ng-excel-builder/src/excel-builder.service.ts [hunk 2]
```diff
@@ -155,27 +160,40 @@ export class ExcelBuilderService {
...
+                    }
+
+                    private getWidthOfString(string: string, fontWeight: keyof WidthOfSymbols): number {
+                        return string
+                            .split('')
+                            .reduce(
+                                (sum: number, symbol: string): number =>
+                                    sum + (input.widthOfSymbols[fontWeight][symbol] ?? this.sizes.fontWidth),
+                                0
+                            );
```
Comment (Schirbak) [inline packages/ng-excel-builder/src/excel-builder.service.ts:196]: i think it's more readable 
```ts
function getWidthOfString(string: string, fontWeight: keyof WidthOfSymbols): number {
    let width: number = 0;
    for (const symbol of string) {
        width += input.widthOfSymbols[fontWeight][symbol] ?? this.sizes.fontWidth;
    }
    return width;
}
```

Comment (splincode) [reply packages/ng-excel-builder/src/excel-builder.service.ts:196]: `input.widthOfSymbols?.[fontWeight]?.[symbol]`

Comment (splincode) [reply packages/ng-excel-builder/src/excel-builder.service.ts:196]: @StepaZol ))

File: packages/ng-excel-builder/src/excel-builder.service.ts [hunk 3]
```diff
@@ -155,27 +160,40 @@ export class ExcelBuilderService {
...
                         }
                     }
 
-                    private calcMaxWidthByEntries(entries: PlainObject[], key: string): number {
-                        const maxLength: number = entries.reduce((length: number, entry: PlainObject): number => {
-                            const currentLength: number = (entry[key]?.toString() ?? '').length;
+                    private calcMaxWidthByEntries(entries: string[], title: string): number {
+                        const titleLength: number = this.getWidthOfString(title, 'bold');
+                        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                        const padding: number = this.sizes.fontWidth * 2;
```
Comment (splincode) [inline packages/ng-excel-builder/src/excel-builder.service.ts:181]: create variable for `2`, what it is? add comments please why

Comment (StepaZol) [reply packages/ng-excel-builder/src/excel-builder.service.ts:181]: I couldn't think of a name for it for a long time yesterday) but now I guessed it is indent measured In symbols

File: packages/ng-excel-builder/src/excel-builder.service.ts (hunk 2 above)
Comment (StepaZol) [reply packages/ng-excel-builder/src/excel-builder.service.ts:196]: I like reduce)

**Review by Schirbak — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Pull request merged**